### PR TITLE
Nag when a build service is used from a task without the corresponding `usesService` declaration

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle test QFL

**Comment by bamboo** [comment]
@bot-gradle test ACC

**Comment by bamboo** [comment]
@bot-gradle test ACC

**Comment by bamboo** [comment]
@bot-gradle test this

File: subprojects/core/src/integTest/groovy/org/gradle/api/services/BuildServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -56,17 +56,29 @@ class BuildServiceIntegrationTest extends AbstractIntegrationSpec {
...
         """
 
+        if (enableFeaturePreview) {
+            settingsFile '''
+                enableFeaturePreview 'STABLE_BUILD_SERVICES'
+            '''
+            executer.expectDocumentedDeprecationWarning(
+                "Build service 'counter' is being used by task ':broken' without the corresponding declaration via 'Task#usesService'. " +
+                    "This will fail with an error in Gradle 8.0. " +
+                    "Max parallel usages constraint on the build service can't be honored. " +
```
Comment (adammurdoch) [inline subprojects/core/src/integTest/groovy/org/gradle/api/services/BuildServiceIntegrationTest.groovy:66]: I think we should add an entry to the upgrade guide for this deprecation and move this kind of detail there, where we can provide more context. We can do this in a later PR, though.

**Review by adammurdoch — APPROVED** [review]

File: subprojects/core/src/integTest/groovy/org/gradle/api/services/BuildServiceIntegrationTest.groovy (hunk 1 above)
Comment (bamboo) [reply subprojects/core/src/integTest/groovy/org/gradle/api/services/BuildServiceIntegrationTest.groovy:66]: 👍

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Pull request merged**